datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -58,21 +59,42 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
     @PostMapping("/bookstores/book")
     protected CollectionModel<EntityModel<Book>> oneBookToManyStores(@RequestBody Book book, @RequestParam List<String> id) throws Exception {
         List<EntityModel<Book>> entityList = new ArrayList<>();
-        for(String storeId: id)
-        {
-            book.setStoreID(Long.parseLong(storeId));
-            if(!this.map.containsKey(Long.parseLong(storeId))) continue;
-            HttpURLConnection con = createConnection(this.map.get(Long.parseLong(storeId)) + "/books", "POST");
+        if (!this.storeRepository.findAll().get(0).getServerId().equals(this.leader.getLeader())) {
```

Comment:
I think it would be very helpful to have a convenience method like `amITheLeader` which encapsulates this logic.